fbw-a380x/src/systems/systems-host/systems/LegacyFuel.ts

```diff
@@ -136,22 +142,30 @@ export class LegacyFuel implements Instrument {
     const dt = this.sysHost.deltaTime;
     const totalDtSinceLastUpdate = this.throttler.canUpdate(dt);
 
-    if (!this.hasInit || totalDtSinceLastUpdate <= 0) {
+    if (!this.hasInit) {
       return;
     }
 
     const onGround = SimVar.GetSimVarValue('SIM ON GROUND', 'bool');
-    if (this.refuelStarted.get()) {
+    if (!this.refuelInProgress && this.refuelStarted.get()) {
       this.refuelInProgress = true;
+
       for (let index = 1; index <= LegacyFuel.NUMBER_OF_TRIGGERS; index++) {
         if (this.triggerStates.get(index).get()) {
           this.keyEventManager.triggerKey('FUELSYSTEM_TRIGGER_OFF', true, index);
         }
       }
-    } else if (this.refuelInProgress) {
+      // starts at 5 since 1-4 are the engine valves controlled by the engine masters
+      // valve 37, 40, 50, 51 should remain unchanged
+      for (let index = 5; index <= LegacyFuel.NUMBER_OF_VALVES; index++) {
+        if (![37, 40, 50, 51].includes(index)) {
```

Comment:
This should only execute once, if refueling just started, right? If it would run every frame, we should avoid the array allocation and rather should check for the inequality with these 4 valves in four statements

Reply:
Could you also explain in the comment why it's these four valves?

Reply:
Yeah, don't allocate objects in onUpdate if at all avoidable. You can make this array a private static readonly member of the class.

Reply:
I was wondering how it would do it internally, I can add it statically ofc